admin-bulk-user-membership/tool/src/main/java/org/sakaiproject/bulk/membership/service/BulkUserMembershipToolService.java

```diff
@@ -0,0 +1,108 @@
+package org.sakaiproject.bulk.membership.service;
+
+import java.util.List;
+import java.util.Collection;
+import java.util.Locale;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.sakaiproject.authz.api.Member;
+import org.sakaiproject.bulk.membership.exception.UsersByEmailException;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class BulkUserMembershipToolService {
+
+    @Autowired
+    private UserDirectoryService userDirectoryService;
+
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+	private PreferencesService preferencesService;
+
+    @Autowired
+    private SiteService siteService;
+
+    public User getUser(String criteria) throws UsersByEmailException {
+        try {
+            User user = userDirectoryService.getUserByEid(criteria);
+            if (user != null) {
+                return user;
+            }
+        } catch (UserNotDefinedException usEx) {
+            Collection<User> users = userDirectoryService.findUsersByEmail(criteria);
+            if (users.size() > 1) {
+                throw new UsersByEmailException(criteria);
+            } else if (!users.isEmpty()) {
+                return users.toArray(new User[users.size()])[0];
+            }
+        }
+        return null;
+    }
+
+    public Site getSite(String id) {
+        if (siteService.siteExists(id)) {
+            try {
+                return siteService.getSite(id);
+            } catch (IdUnusedException idEx) { }
+        }
+        return null;
+    }
+
+    public void applyAction(String action, Site site, String userCriteria, String role) throws Exception {
+        String userId = getUser(userCriteria).getId();
+        Member member = site.getMember(userId);
+        switch (action) {
+            case "add":
```

Comment:
Use an enum please, and define a default case with an error.

enum ACTIONS = {ADD, REMOVE, ACTIVATE, DEACTIVATE}

Reply:
Hello  @mpellicer , here it is!!

common_title=Kudeatu partaidetza masiboki

automembership.wizard.step1=1 urratsa: Ekintza hautaketa
automembership.wizard.step2=2 urratsa: Berrespena
automembership.wizard.step3=3 urratsa: Emaitzak

fail_user= Erabiltzaile hau ez dago plataforman: {0} 
fail_user_duplicated= Bi erabiltzailek (edo gehiagok) dute email berdina: {0}
fail_site= Gune hau ez da existitzen: {0}

no_users=Erabiltzailerik ez duzu hautatu
no_sites=Gunerik ez duzu hautatu
no_action=Ekintzarik ez duzu hautatu

box.users.list.title=Partaideak
box.users.list.subtitle=Partaideen identifikatzaileen zerrenda
box.sites.list.title=Guneak
box.sites.list.subtitle=Guneen identifikatzaileen zerrenda
box.succesfull.list.title=erabiltzaileak ongi {0} dira gune hauetan: 
box.fail.list.title=Hurrengo erabiltzaile hauek izan dira {0}:

action.selection.title=Ekintza burutzeko: 
action.selection.add=Gehitu partaideak
action.selection.remove=Ezabatu partaideak
action.selection.activate=Aktibatu partaideak
action.selection.deactivate=Desaktibatu partaideak

site.title=Guneak
role.title=Rola
button.cancel=Utzi
button.continue=Jarraitu
button.finish=Bukatu
button.back=Atzera
button.another=Beste ekintza
select.role.title= Aukeratu zer rola izango duten erabiltzaile guzti horiek guneetan
id_title={0} ({1})

add=GEHITU
remove= EZABATU
activate= AKTIBATU
deactivate= DESAKTIBATU

automembership.title.step1=Behean dauden ekintzen artean bat hautatu behar duzu. Ezkerrean dauden erabiltzaileei ekintza hori aplikatuko zaie, eta ekintza hori gertatuko da eskubian dauden guneetan. 
automembership.title.step2=Ekintza hau burutuko da: erabiltzaileak {0}. Eskubian dauden guneetan gertatuko da ekintza.
automembership.title.step3=Emaitzen pantaila